ComplexDialogShortcutIT::dialogOpenedWithListenOnShortcut_dialogReopened_oldShortcutStillWorks is flaky

ComplexDialogShortcutIT::dialogOpenedWithListenOnShortcut_dialogReopened_oldShortcutStillWorks fails without any reasonable cause in `Flow Tomcat Integration Tests (2.5)`:

https://bender.vaadin.com/viewLog.html?buildId=221431&buildTypeId=Flow_Flow25_Nightly_FlowTomcatIntegrationTests25

![Screenshot 2021-03-30 at 10 04 31](https://user-images.githubusercontent.com/61410877/112947401-66e23500-913f-11eb-928c-1424cf6953fa.png)

```
java.lang.RuntimeException: dialogOpenedWithListenOnShortcut_dialogReopened_oldShortcutStillWorks[ANY_Chrome_](com.vaadin.flow.uitest.ui.ComplexDialogShortcutIT): run failed 2 times
	at com.vaadin.testbench.RetryRule$1.evaluate(RetryRule.java:69)
	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
	at org.junit.runners.BlockJUnit4ClassRunner$1.evaluate(BlockJUnit4ClassRunner.java:100)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:366)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:103)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:63)
	at org.junit.runners.ParentRunner$4.run(ParentRunner.java:331)
	at org.apache.maven.surefire.junitcore.pc.Scheduler$1.run(Scheduler.java:410)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
Caused by: org.junit.ComparisonFailure: Invalid latest event with 0::DIALOG999 expected:<[4-DIALOG999]> but was:<[3-ui-button]>
	at org.junit.Assert.assertEquals(Assert.java:117)
	at com.vaadin.flow.uitest.ui.DialogShortcutIT.validateShortcutEvent(DialogShortcutIT.java:205)
	at com.vaadin.flow.uitest.ui.DialogShortcutIT.validateLatestDialogShortcut(DialogShortcutIT.java:192)
	at com.vaadin.flow.uitest.ui.DialogShortcutIT.dialogOpenedWithListenOnShortcut_dialogReopened_oldShortcutStillWorks(DialogShortcutIT.java:104)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
	at com.vaadin.testbench.parallel.ParallelRunner$TBMethod.invokeExplosively(ParallelRunner.java:494)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)
	at com.vaadin.testbench.parallel.ParallelRunner$1.evaluate(ParallelRunner.java:468)
	at org.junit.rules.TestWatcher$1.evaluate(TestWatcher.java:61)
	at org.junit.rules.TestWatcher$1.evaluate(TestWatcher.java:61)
	at com.vaadin.testbench.RetryRule$1.evaluate(RetryRule.java:59)
	... 12 more
------- Stderr: -------
Mar 29, 2021 10:03:58 PM org.openqa.selenium.remote.ProtocolHandshake createSession
INFO: Detected dialect: W3C
Mar 29, 2021 10:03:59 PM com.vaadin.testbench.ScreenshotOnFailureRule failed
INFO: Error screenshot written to: /opt/agent/work/1911471312825fc3/flow-tests/servlet-containers/tomcat85/error-screenshots/dialogOpenedWithListenOnShortcut_dialogReopened_oldShortcutStillWorks[ANY_Chrome_](com.vaadin.flow.uitest.ui.ComplexDialogShortcutIT).png
Mar 29, 2021 10:04:00 PM org.openqa.selenium.remote.ProtocolHandshake createSession
INFO: Detected dialect: W3C
Mar 29, 2021 10:04:02 PM com.vaadin.testbench.ScreenshotOnFailureRule failed
INFO: Error screenshot written to: /opt/agent/work/1911471312825fc3/flow-tests/servlet-containers/tomcat85/error-screenshots/dialogOpenedWithListenOnShortcut_dialogReopened_oldShortcutStillWorks[ANY_Chrome_](com.vaadin.flow.uitest.ui.ComplexDialogShortcutIT).png
```


```
[01:06:35]	[Step 5/5] [INFO] Running com.vaadin.flow.uitest.ui.ComplexDialogShortcutIT
[01:06:35]	[Step 5/5] [ERROR] Tests run: 7, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 6.555 s <<< FAILURE! - in com.vaadin.flow.uitest.ui.ComplexDialogShortcutIT
[01:06:35]	[Step 5/5] [ERROR] dialogOpenedWithListenOnShortcut_dialogReopened_oldShortcutStillWorks[ANY_Chrome_](com.vaadin.flow.uitest.ui.ComplexDialogShortcutIT)  Time elapsed: 6.386 s  <<< ERROR!
[01:06:35]	[Step 5/5] java.lang.RuntimeException: dialogOpenedWithListenOnShortcut_dialogReopened_oldShortcutStillWorks[ANY_Chrome_](com.vaadin.flow.uitest.ui.ComplexDialogShortcutIT): run failed 2 times
[01:06:35]	[Step 5/5] 	at com.vaadin.testbench.RetryRule$1.evaluate(RetryRule.java:69)
[01:06:35]	[Step 5/5] 	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
[01:06:35]	[Step 5/5] 	at org.junit.runners.BlockJUnit4ClassRunner$1.evaluate(BlockJUnit4ClassRunner.java:100)
[01:06:35]	[Step 5/5] 	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:366)
[01:06:35]	[Step 5/5] 	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:103)
[01:06:35]	[Step 5/5] 	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:63)
[01:06:35]	[Step 5/5] 	at org.junit.runners.ParentRunner$4.run(ParentRunner.java:331)
[01:06:35]	[Step 5/5] 	at org.apache.maven.surefire.junitcore.pc.Scheduler$1.run(Scheduler.java:410)
[01:06:35]	[Step 5/5] 	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
[01:06:35]	[Step 5/5] 	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
[01:06:35]	[Step 5/5] 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
[01:06:35]	[Step 5/5] 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
[01:06:35]	[Step 5/5] 	at java.lang.Thread.run(Thread.java:748)
[01:06:35]	[Step 5/5] Caused by: org.junit.ComparisonFailure: Invalid latest event with 0::DIALOG999 expected:<[4-DIALOG999]> but was:<[3-ui-button]>
[01:06:35]	[Step 5/5] 	at org.junit.Assert.assertEquals(Assert.java:117)
[01:06:35]	[Step 5/5] 	at com.vaadin.flow.uitest.ui.DialogShortcutIT.validateShortcutEvent(DialogShortcutIT.java:205)
[01:06:35]	[Step 5/5] 	at com.vaadin.flow.uitest.ui.DialogShortcutIT.validateLatestDialogShortcut(DialogShortcutIT.java:192)
[01:06:35]	[Step 5/5] 	at com.vaadin.flow.uitest.ui.DialogShortcutIT.dialogOpenedWithListenOnShortcut_dialogReopened_oldShortcutStillWorks(DialogShortcutIT.java:104)
[01:06:35]	[Step 5/5] 	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
[01:06:35]	[Step 5/5] 	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
[01:06:35]	[Step 5/5] 	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
[01:06:35]	[Step 5/5] 	at java.lang.reflect.Method.invoke(Method.java:498)
[01:06:35]	[Step 5/5] 	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
[01:06:35]	[Step 5/5] 	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
[01:06:35]	[Step 5/5] 	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
[01:06:35]	[Step 5/5] 	at com.vaadin.testbench.parallel.ParallelRunner$TBMethod.invokeExplosively(ParallelRunner.java:494)
[01:06:35]	[Step 5/5] 	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
[01:06:35]	[Step 5/5] 	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)
[01:06:35]	[Step 5/5] 	at com.vaadin.testbench.parallel.ParallelRunner$1.evaluate(ParallelRunner.java:468)
[01:06:35]	[Step 5/5] 	at org.junit.rules.TestWatcher$1.evaluate(TestWatcher.java:61)
[01:06:35]	[Step 5/5] 	at org.junit.rules.TestWatcher$1.evaluate(TestWatcher.java:61)
[01:06:35]	[Step 5/5] 	at com.vaadin.testbench.RetryRule$1.evaluate(RetryRule.java:59)
[01:06:35]	[Step 5/5] 	... 12 more

01:06:35]	[Step 5/5] [INFO] Results:
[01:06:35]	[Step 5/5] [INFO] 
[01:06:35]	[Step 5/5] [ERROR] Errors: 
[01:06:35]	[Step 5/5] [ERROR]   ComplexDialogShortcutIT.dialogOpenedWithListenOnShortcut_dialogReopened_oldShortcutStillWorks[ANY_Chrome_] » Runtime
[01:06:35]	[Step 5/5] [INFO] 
[01:06:35]	[Step 5/5] [ERROR] Tests run: 311, Failures: 0, Errors: 1, Skipped: 4
```


Screenshots:

flow-tests/servlet-containers/tomcat85/errorscreenshots/dialogOpenedWithListenOnShortcut_dialogReopened_oldShortcutStillWorks%5BANY_Chrome_%5D(com.vaadin.flow.uitest.ui.ComplexDialogShortcutIT).png

![image](https://user-images.githubusercontent.com/61410877/112947601-a872e000-913f-11eb-9e79-46dc976702cb.png)

flow-tests/servlet-containers/tomcat85/error-screenshots/dialogWithShortcutListenOnAndPreserveOnRefresh_refreshWhenDialogOpened_shortcutScopingWorks%5BANY_Chrome_%5D(com.vaadin.flow.uitest.ui.ComplexDialogShortcutIT).png

![image](https://user-images.githubusercontent.com/61410877/112947692-c8a29f00-913f-11eb-9d68-0bcb31869df0.png)

flow-tests/servlet-containers/tomcat85/error-screenshots/pushUpdatesEmbeddedWebComponent%5BANY_Chrome_%5D(com.vaadin.flow.uitest.ui.PushLongPollingUpdateDivIT).png

![image](https://user-images.githubusercontent.com/61410877/112947738-d9531500-913f-11eb-9fe7-243081fda570.png)

--- a/flow-tests/test-root-context/src/test/java/com/vaadin/flow/uitest/ui/ComplexDialogShortcutIT.java
+++ b/flow-tests/test-root-context/src/test/java/com/vaadin/flow/uitest/ui/ComplexDialogShortcutIT.java
@@ -1,12 +1,10 @@
 package com.vaadin.flow.uitest.ui;
 
-import org.junit.Ignore;
 import org.junit.Test;
 import org.openqa.selenium.By;
 
 import static com.vaadin.flow.uitest.ui.DialogShortcutView.REUSABLE_DIALOG_ID;
 
-@Ignore("Flaky test #10481, #10487, #10491")
 public class ComplexDialogShortcutIT extends DialogShortcutIT {
 
     @Test

--- a/flow-tests/test-root-context/src/test/java/com/vaadin/flow/uitest/ui/DialogShortcutIT.java
+++ b/flow-tests/test-root-context/src/test/java/com/vaadin/flow/uitest/ui/DialogShortcutIT.java
@@ -9,14 +9,12 @@ import com.vaadin.flow.testutil.ChromeBrowserTest;
 import com.vaadin.testbench.TestBenchElement;
 import org.junit.Assert;
 import org.junit.Before;
-import org.junit.Ignore;
 import org.junit.Test;
 import org.openqa.selenium.By;
 import org.openqa.selenium.WebElement;
 
 import static com.vaadin.flow.uitest.ui.DialogShortcutView.REUSABLE_DIALOG_ID;
 
-@Ignore("Flaky test #10481, #10487, #10491")
 public class DialogShortcutIT extends ChromeBrowserTest {
 
     private TestBenchElement eventLog;
@@ -201,8 +199,8 @@ public class DialogShortcutIT extends ChromeBrowserTest {
 
     private void validateShortcutEvent(int indexFromTop, int eventCounter,
             String eventSourceId) {
-        final WebElement latestEvent = eventLog.findElements(By.tagName("div"))
-                .get(indexFromTop);
+        final WebElement latestEvent = eventLog.findElement(
+                By.xpath(String.format("div[%d]", indexFromTop + 1)));
         Assert.assertEquals(
                 "Invalid latest event with " + indexFromTop + ":" + ":"
                         + eventSourceId,

--- a/flow-tests/test-root-context/src/test/java/com/vaadin/flow/uitest/ui/DialogShortcutIT.java
+++ b/flow-tests/test-root-context/src/test/java/com/vaadin/flow/uitest/ui/DialogShortcutIT.java
@@ -199,8 +199,8 @@ public class DialogShortcutIT extends ChromeBrowserTest {
 
     private void validateShortcutEvent(int indexFromTop, int eventCounter,
             String eventSourceId) {
-        final WebElement latestEvent = eventLog.findElement(
-                By.xpath(String.format("div[%d]", indexFromTop + 1)));
+        final WebElement latestEvent = waitUntil(driver -> eventLog.findElement(
+                By.xpath(String.format("div[%d]", indexFromTop + 1))));
         Assert.assertEquals(
                 "Invalid latest event with " + indexFromTop + ":" + ":"
                         + eventSourceId,

